### PR TITLE
Migrate type from String[] to LabelData[] (2022)

### PR DESCRIPTION
**Pull request by ailZhou**
### Description

This is the same [migration done here](https://github.com/Enterprise-CMCS/macpro-mdct-qmr/pull/2194) but for 2022. It was the mostly the exact same updates but to the files for 2022.

Only two difference is that OEV-AD was added in 2022 and CCW-AD validation has a different pass in value than 2021 CCW-AD

### Related ticket(s)

CMDCT-3497

---
### How to test

1. Sign into QMR, [stateuser1@test.com](mailto:stateuser1@test.com)
2. Select reporting year 2022
3. Measure recommended to test, AIF-HH, IU-HH, IET-AD, PCR-AD, PCC-AD, BCS-AD and/or WCC-CH
4. Fill out the measure as you would a user, entering a performance measure and an OMS.
5. Validate and make sure there's no errors

### Notes

N/A

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by BearHanded — APPROVED** [review]
That's a lot of types

File: services/ui-src/src/measures/2022/rateLabelText.ts [hunk 1]
```diff
@@ -1660,13 +1660,19 @@ export const data = {
...
+    const qualifiers: LabelData[] = data[measure].qualifiers.map((item) => ({
+        id: cleanString(item.label), //for some reason the system would create the id from the label instead of using the id key that existed
+        label: item.label,
+        text: item.text,
+    }));
+    const categories: LabelData[] = data[measure].categories.map((item) => ({
+        id: cleanString(item.label), //for some reason the system would create the id from the label instead of using the id key that existed
+        label: item.label,
+        text: item.text,
+    }));
```
Comment (benmartin-coforma) [inline services/ui-src/src/measures/2022/rateLabelText.ts:1672]: QUESTION: I don't understand what's going on here. It seems like lines 1663 and 1668 could be replaced with
```ts
    id: item.id,
```
which would mean that the whole block could be replaced with
```ts
  const qualifiers = data[measure].qualifiers;
  const categories = data[measure].categories;
```
which would mean that the whole function could be replaced with
```
  return data[measure];
```

I must be missing something, but I don't know what it is.

**Review by benmartin-coforma — COMMENTED** [review]
I love this PR. Every step away from regex-based behavior towards structured data brings me joy.

File: services/ui-src/src/measures/2022/rateLabelText.ts (hunk 1 above)
Comment (ailZhou) [reply services/ui-src/src/measures/2022/rateLabelText.ts:1672]: you are 100% correct but the thing is, i don't trust that all the ids value are equivalent to cleanString(item.label). It's my lack of faith in this app more than anything else. i could probably check with a loop, but i would never believe the results isn't lying to me 🫠 .

File: services/ui-src/src/measures/2022/shared/globalValidations/dataDrivenTools.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import * as DC from "dataConstants";
 import * as Types from "measures/2022/shared/CommonQuestions/types";
 import { DataDrivenTypes as DDT } from "measures/2022/shared/CommonQuestions/types";
-import { cleanString } from "utils";
+import { LabelData, cleanString } from "utils";
```
[the remark this answers is not in the document]
Comment (ailZhou) [reply services/ui-src/src/measures/2022/shared/globalValidations/dataDrivenTools.ts:4]: yes! removing cleanString is definitely a goal of mine for the future. for now, the first step is just trimming the excess code.

File: services/ui-src/src/measures/2022/shared/globalValidations/validateNumeratorsLessThanDenominators/index.test.ts [hunk 3]
```diff
@@ -138,12 +149,17 @@ describe("Testing Numerator Less Than Denominator", () => {
...
       ]);
     });
   });
 
   it("Error message text should match provided errorMessage", () => {
+    const utils = require("utils/getLabelText");
+    jest.spyOn(utils, "isLegacyLabel").mockImplementation(() => {
+      return true;
+    });
+
```
[the remark this answers is not in the document]
Comment (ailZhou) [reply services/ui-src/src/measures/2022/shared/globalValidations/validateNumeratorsLessThanDenominators/index.test.ts:162]: that's strange, its failing for me locally when removed?

[the remark this answers is not in the document]
Comment (benmartin-coforma) [reply services/ui-src/src/measures/2022/shared/globalValidations/validateNumeratorsLessThanDenominators/index.test.ts:162]: That is odd... 🤔 but it's not worth holding up this PR for.

**Review by benmartin-coforma — APPROVED** [review]

**Review by BearHanded — APPROVED** [review]

**Pull request merged**